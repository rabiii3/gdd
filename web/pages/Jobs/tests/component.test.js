import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore  from '../../../store/configureStore.test';
import { JobDescription } from '..';

describe( 'app | pages | JobDescription | component', () => {
  it('should match snapshot', () => {
    const state = {
      auth:{
        user:{
          _id: '123',
          firstname: 'user',
          lastname: 'user',
          locale: 'en',
        }
      },
      skills:{
        data:{
          '5d035ef378c332001b67bada':{
            _id: '5d035ef378c332001b67bada',
            label: 'DevOps',
          },
          '5d035ef378c332001b67badb':{
            _id: '5d035ef378c332001b67badb',
            label: 'React JS',
          }
        }
      },
      jobDescription:{
        data:{
          '0':{
            _id: '0',
            title: 'required js dev',
            content: 'some content',
            skills: ['5d035ef378c332001b67bada', '5d035ef378c332001b67badb'],
            createdBy: '456',
            createdAt: '',
            updatedBy: '456',
            updatedAt: '',
            tenantId: '333'
          }
        },
        sort:{
          sortBy: 'TITLE',
          direction: 'desc',
        }
      },
      people:{
        data:{
          '456':{
            _id: '456',
            roles: ['admin'],
            firstname: 'test',
            lastname: 'tester',
          }
        }
      }
    };
    const props = {
      classes:{}, 
      history:{}, 
      routes:{}, 
      user:{}, 
      sortedJobs:[], 
      skills:{}, 
      selectedJob:"", 
      handleSelectedJob:jest.fn(), 
      sortJobs:jest.fn(),
      deleteJob:jest.fn(),
      sortMode:{},
      people:{},
    };
    const store = configureStore(state);
      const wrapper = shallow(
        <Provider store={store}>
          <JobDescription {...props} />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
  });
});