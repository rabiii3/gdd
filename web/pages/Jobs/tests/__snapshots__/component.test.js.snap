// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app | pages | JobDescription | component should match snapshot 1`] = `
<Component
  classes={Object {}}
  deleteJob={[MockFunction]}
  handleSelectedJob={[MockFunction]}
  history={Object {}}
  people={Object {}}
  routes={Object {}}
  selectedJob=""
  skills={Object {}}
  sortJobs={[MockFunction]}
  sortMode={Object {}}
  sortedJobs={Array []}
  user={Object {}}
/>
`;
